ClosedXML/Excel/XLWorkbook_Load.cs

```diff
@@ -1004,22 +1004,73 @@ private void LoadDefinedNames(Workbook workbook)
 
                 if (name == "_xlnm.Print_Area")
                 {
-                    var fixedNames = validateDefinedNames(definedName.Text.Split(','));
-                    foreach (string area in fixedNames)
+                    try
+                    {
+                        // NOTE: try previous simpler implementation first in order to still be able to adapt ranges of print areas via code
+                        // NOTE: collect all printAreas first since in the catch block all expressions are parsed and added again if there is a failure in the parsing here
+                        var printAreasToAdd = new List<(IXLPrintAreas, string)>();
+                        var fixedNames = validateDefinedNames(definedName.Text.Split(','));
+                        foreach (string area in fixedNames)
+                        {
+                            if (area.Contains("["))
+                            {
+                                var worksheetForAdding = WorksheetsInternal.FirstOrDefault<XLWorksheet>(w => w.SheetId == (localSheetId + 1));
+
+                                if (worksheetForAdding.Range(area) == null)
+                                    throw new ArgumentNullException();
+
+                                printAreasToAdd.Add((worksheetForAdding.PageSetup.PrintAreas, area));
+                            }
+                            else
+                            {
+                                ParseReference(area, out String sheetName, out String sheetArea);
+
+                                if (!(sheetArea.Equals("#REF") || sheetArea.EndsWith("#REF!") || sheetArea.Length == 0 || sheetName.Length == 0))
+                                {
+                                    var worksheetForAdding = WorksheetsInternal.Worksheet(sheetName);
+
+                                    if (worksheetForAdding.Range(sheetArea) == null)
+                                        throw new ArgumentNullException();
+
+                                    printAreasToAdd.Add((worksheetForAdding.PageSetup.PrintAreas, sheetArea));
+                                }
+                            }
+                        }
+
+                        foreach (var printAreaToAdd in printAreasToAdd)
+                            printAreaToAdd.Item1.Add(printAreaToAdd.Item2);
+                    }
+                    catch (Exception)
                     {
-                        if (area.Contains("["))
+                        var formula = CalcEngine.Parse(definedName.Text);
```

Comment:
Please don't use `Formula` if possible. That was a valid approach back when we used _XLParser_ to parse formulas, but back then it took ~150 μs to parse formula IIRC (i.e. 10k would take 1.5 second). Current approach is to do the work while parsing without building the tree in memory (`IAstVisitor` ect.). It;s simpler (one representation) and more memory efficient. Formula and materialized tree is going the way of the dodo.

Reply:
I see. I changed the PrintArea to hold a string as you suggested in another comment. I do not quite know if anything else has to be done to be honest.